woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+                for sample, map_ in rmap.items():
+                    for rank in ranks:
+
+                        # call assignment workflow
+                        assign_rmap(map_, data, rank, sample, **kwargs)
+
+        click.echo(' Done.')
+        click.echo(f'Query sequences: {n}.')
+
+    click.echo('Task completed.')
+    return data
+
+
+def parse_samples(path_:  str,
+                  ext:    str = None,
+                  ids:   list = None,
+                  demux: bool = None) -> (list, list or dict, bool):
+    """Determine sample IDs, aligment files, and multiplex status.
+
+    Parameters
+    ----------
+    path_ : str
+        Path to a file or a directory.
+    ext : str, optional
+        Filename extension.
+    ids : iterable of str, optional
+        List of sample IDs.
+    demux : bool, optional
+        Whether perform demultiplexing.
+
+    Returns
+    -------
+    list, list or dict, bool
+        Sample IDs to include.
+        Filepaths if demultiplexing, or filepath to sample ID map if not.
+        Whether perform demultiplexing.
+    """
+    # read sample Ids
+    if ids:
+        samples = read_ids(ids)
+        click.echo(f'Samples to include: {len(samples)}.')
+    else:
+        samples = None
+
+    errmsg = 'Provided sample IDs and actual files are inconsistent.'
+
+    # path is an alignment file
+    if isfile(path_):
+
+        # turn on demultiplexing if not decided
+        demux = demux is not False
+        if demux:
+            files = [path_]
+
+        # validate with given sample Ids
+        else:
+            sample = path2stem(path_, ext)
+            if samples and samples != [sample]:
+                raise ValueError(errmsg)
+            files = {path_: sample}
+            samples = [sample]
+
+    # path is a directory of alignment files
+    elif isdir(path_):
+
+        # turn off demultiplexing if not decided
+        demux = demux or False
+
+        # get a map of plausible sample Ids to files
+        map_ = id2file_map(path_, ext, not demux and samples)
+        if len(map_) == 0:
+            raise ValueError('No valid file found in directory.')
+        if demux:
+            files = sorted([join(path_, x) for x in map_.values()])
+
+        # validate with given sample Ids
+        else:
+            if not samples:
+                samples = sorted(map_.keys())
+            elif len(map_) < len(samples):
+                raise ValueError(errmsg)
+            files = {join(path_, map_[x]): x for x in samples}
+
+    else:
+        raise ValueError(f'"{path_}" is not a valid file or directory.')
+
+    click.echo(f'Alignment files to read: {len(files)}.')
+    click.echo(f'Demultiplexing: {"on" if demux else "off"}.')
+    return samples, files, demux
+
+
+def build_align_proc(coords_fp: str = None,
+                     overlap:   int = None) -> object:
+    """Build alignment processor.
+
+    Parameters
+    ----------
+    coords_fp : str, optional
```

Comment:
```suggestion
    coordinates_fp : str, optional
```
and subsequent changes in function. Makes code more readable.

Reply:
I guess `coords` is a very common abbr. For example: https://www.w3schools.com/tags/att_a_coords.asp. `coordinates_fp` is bit too long as a command-line argument as well as a variable.

Reply:
I usually tend toward the side of "too much detail is better than too little detail" (and I think the qiime developers tend to side with this as well, notice that most core qiime packages have almost no abbreviations). But understood. Resolved.